### PR TITLE
Fix prompt behavior for native requests

### PR DESCRIPTION
**Pull request by tnorling**
- Fixes bug where calling a silent API with prompt != none throws and instead overwrites prompt to always be none for silent calls
- Fixes bug where prompt=login skips the native broker and goes directly to web

### CODE REVIEW

File: lib/msal-browser/src/interaction_client/NativeInteractionClient.ts [hunk 1]
```diff
@@ -319,17 +319,34 @@ export class NativeInteractionClient extends BaseInteractionClient {
         const scopeSet = new ScopeSet(scopes);
         scopeSet.appendScopes(OIDC_DEFAULT_SCOPES);
 
-        if (request.prompt) {
+        const prompt = () => {
```
Comment (jasonnutter) [inline lib/msal-browser/src/interaction_client/NativeInteractionClient.ts]:
```suggestion
        const getPrompt = () => {
```

or similar?

File: lib/msal-browser/src/interaction_client/NativeInteractionClient.ts [hunk 2]
```diff
@@ -319,17 +319,34 @@ export class NativeInteractionClient extends BaseInteractionClient {
...
+                case ApiId.ssoSilent:
+                case ApiId.acquireTokenSilent_silentFlow:
+                    this.logger.trace("initializeNativeRequest: silent request sets prompt to none");
+                    return PromptValue.NONE;
+                default:
+                    break;
+            }
+
+            // If request is interactive, check if prompt is allowed to go directly to native broker
+            if (!request.prompt) {
```
Comment (jasonnutter) [inline lib/msal-browser/src/interaction_client/NativeInteractionClient.ts:334]: Not sure I follow this given the comment. Is this saying request.prompt being falsey means the request shuold go to the broker?

File: lib/msal-browser/src/interaction_client/NativeInteractionClient.ts (hunk 1 above)
Comment (tnorling) [reply lib/msal-browser/src/interaction_client/NativeInteractionClient.ts]: Done

File: lib/msal-browser/src/interaction_client/NativeInteractionClient.ts (hunk 2 above)
Comment (tnorling) [reply lib/msal-browser/src/interaction_client/NativeInteractionClient.ts:334]: Yes, rephrased to clarify

**Review by jasonnutter — APPROVED** [review]

**Review by peterzenz — APPROVED** [review]

**Pull request merged**